### PR TITLE
Created a tilemap example for testing texture atlases

### PR DESCRIPTION
**Pull request by Raelr**
### Description

This PR adds a new example to the project that's meant to demonstrate some 2D tilemap capabilities. The idea is that this example will be a test-bed for 2D features across the project and POTENTIALLY become the foundation for an actual game.

### The PR has been...
- [x] provided a reasonable name that is not just the branch name (e.g "Added Vulkan render delegate")
- [x] linked to its related issue
- [x] assigned a reviewer from the team
- [x] labelled appropriately

### The code has been...
- [x] made mergable and free of conflicts in relation to `master` *(according to GitHub)*
- [x] tested in a packaged state using the `package` targets
- [x] pulled to the reviewer's machine and reasonably tested

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -78,18 +79,26 @@ $(exampleGameApp): buildFlags $(renderLib) $(coreLib)
...
 
+tilemapapp: $(exampleTilemapApp)
+
 package-gameapp: gameapp
 	"$(MAKE)" package -C $(examplesDir)/game CXXFLAGS="$(CXXFLAGS)"
 
 package-renderapp: renderapp
 	"$(MAKE)" package -C $(examplesDir)/render CXXFLAGS="$(CXXFLAGS)"
 
+package-tilemap: tilemapapp
```
Comment (jonjondev) [inline Makefile:99]:
```suggestion
package-tilemapapp: tilemapapp
```

File: Makefile [hunk 2]
```diff
@@ -47,15 +47,16 @@ export renderLib := $(libDir)/librender.a
...
 export exampleRenderApp := $(binDir)/examples/render/build/app
+export exampleTilemapApp := $(binDir)/examples/tilemap/build/app
 
 # Set build vars
 export compileFlags := -Wall -std=c++17
 export linkFlags += -L $(libDir)
 buildFlagsFile:=.buildflags
 
-.PHONY: all testapp gameapp renderapp package-gameapp package-renderapp buildFlags clean format
+.PHONY: all testapp gameapp renderapp tilemapapp package-gameapp package-renderapp package-tilemap buildFlags clean format
```
Comment (jonjondev) [inline Makefile:57]:
```suggestion
.PHONY: all testapp gameapp renderapp tilemapapp package-gameapp package-renderapp package-tilemapapp buildFlags clean format
```

**Review by jonjondev — APPROVED** [review]
Looks good. Seems like it was pretty straight forward to add.

**Comment by jonjondev** [comment]
Make sure to add a description to the PR explaining what the change is for.

File: Makefile (hunk 1 above)
Comment (Raelr) [reply Makefile:99]: Ah damn, I thought I caught that!

**Pull request merged**